tests/suites/test_suite_ssl.function

```diff
@@ -1,6 +1,196 @@
 /* BEGIN_HEADER */
 #include <mbedtls/ssl.h>
 #include <mbedtls/ssl_internal.h>
+
+/*
+ * Helper function setting up inverse record transformations
+ * using given cipher, hash, EtM mode, authentication tag length,
+ * and version.
+ */
+
+#define CHK( x )                                \
+    do                                          \
+    {                                           \
+        if( !( x ) )                            \
+            return( 1 );                        \
+    } while( 0 )
+
+
+static int build_transforms( mbedtls_ssl_transform *t_in,
+                             mbedtls_ssl_transform *t_out,
+                             int cipher_type, int hash_id,
+                             int etm, int tag_mode, int ver )
+{
+    mbedtls_cipher_info_t const *cipher_info;
+    mbedtls_md_info_t const *md_info;
+
+    size_t keylen, maclen, ivlen;
+    unsigned char *key0, *key1, *md0, *md1;
+    unsigned char iv_enc[16], iv_dec[16];
+
+    /* Pick cipher */
+    cipher_info = mbedtls_cipher_info_from_type( cipher_type );
+    CHK( cipher_info != NULL );
+    CHK( cipher_info->iv_size <= 16 );
+    CHK( cipher_info->key_bitlen % 8 == 0 );
+
+    /* Pick keys */
+    keylen = cipher_info->key_bitlen / 8;
+    CHK( ( key0 = malloc( keylen ) ) != NULL );
+    CHK( ( key1 = malloc( keylen ) ) != NULL );
+    memset( key0, 0x1, keylen );
+    memset( key1, 0x2, keylen );
+
+    /* Pick hash */
+    md_info = mbedtls_md_info_from_type( hash_id );
+    CHK( md_info != NULL );
+
+    /* Pick hash keys */
+    maclen = mbedtls_md_get_size( md_info );
+    CHK( ( md0 = malloc( maclen ) ) != NULL );
+    CHK( ( md1 = malloc( maclen ) ) != NULL );
+    memset( md0, 0x5, maclen );
+    memset( md1, 0x6, maclen );
+
+    /* Pick IV's (regardless of whether they
+     * are being used by the transform). */
+    ivlen = cipher_info->iv_size;
+    memset( iv_enc, 0x3, sizeof( iv_enc ) );
+    memset( iv_dec, 0x4, sizeof( iv_dec ) );
+
+    /*
+     * Setup transforms
+     */
+
+    t_out->encrypt_then_mac = etm;
```

Comment:
This member only exists if `defined(MBEDTLS_SSL_ENCRYPT_THEN_MAC)`. (This issue or another instance of it was caught by Travis CI.)

Reply:
Fixed in 0d677f0.